src/sentry/api/endpoints/group_events.py

```diff
@@ -99,33 +112,15 @@ def _get_events_snuba(self, request: Request, group, environments, query, tags,
             paginator=GenericOffsetPaginator(data_fn=data_fn),
         )
 
-    def _get_search_query_and_tags(self, request: Request, group, environments=None):
+    def _get_search_query(
+        self, request: Request, group: Group, environments: Sequence[Environment]
+    ) -> Optional[str]:
         raw_query = request.GET.get("query")
 
         if raw_query:
             query_kwargs = parse_query([group.project], raw_query, request.user, environments)
-            query = query_kwargs.pop("query", None)
-            tags = query_kwargs.pop("tags", {})
+            query = cast(str, query_kwargs.pop("query", None))
         else:
             query = None
-            tags = {}
 
-        if environments:
-            env_names = {env.name for env in environments}
-            if "environment" in tags:
-                # If a single environment was passed as part of the query, then
-                # we'll just search for that individual environment in this
-                # query, even if more are selected.
-                if tags["environment"] not in env_names:
-                    # An event can only be associated with a single
-                    # environment, so if the environments associated with
-                    # the request don't contain the environment provided as a
-                    # tag lookup, the query cannot contain any valid results.
-                    raise NoResults
-            else:
-                # XXX: Handle legacy backends here. Just store environment as a
-                # single tag if we only have one so that we don't break existing
-                # usage.
-                tags["environment"] = list(env_names) if len(env_names) > 1 else env_names.pop()
-
-        return query, tags
```

Comment:
These tags were never used and are a throwback to before we used snuba. We do all this filtering as part of snuba now, and they're not actually used